test/promise.js

```diff
@@ -152,4 +152,37 @@ describe('Promise', function () {
       }, 50);
     });
   });
-}); 
+  describe('Promise.prototype.then', function() {
+    var spy1 = sinon.spy();
+
+    function Subclass() {
+      spy1();
+      Promise.apply(this, arguments);
+    }
+
+    Object.setPrototypeOf(Subclass.prototype, Promise.prototype);
+
+    Subclass.prototype.then = function() {
+      return Promise.prototype.then.apply(this, arguments);
+    };
+
+    it('subclassed Promise resolves to subclass', function() {
+      var prom = new Subclass(function(resolve) {
+        resolve();
+      }).then(function() {
+        assert(spy1.called);
+        assert(prom instanceof Subclass);
+      });
```

Comment:
These two assert can be outside the `then`. Use the exact number, `.calledTwice`. Use `assert.instanceOf` it will give better error messages.
```js
var prom = new Subclass(function(resolve) {
  resolve();
}).then(function() {});
assert(spy1.calledTwice);
assert.insanceOf(prom, Subclass);
```